### PR TITLE
Clean up moved plotting callbacks

### LINKED ISSUE #1103: redundant matplotlib in bluesky.callbacks.core
Whilst testing the databroker-elasticsearch callback prototype I found that I need matplotlib to import bluesky callbacks.  There was a prior commit a36eb964 to make `matplotlib` optional, but the code still requires matplotlib later on in the mpl_plotting module.

As of df6b93641c93de157165ebe1559c68e85b0f29b2 `pyflakes` report quite a few unused imports in `callbacks.core`:
```sh
$ pyflakes core.py
core.py:6: 'collections.ChainMap' imported but unused
core.py:10: 'numpy as np' imported but unused
core.py:17: 'matplotlib' imported but unused
core.py:19: '.mpl_plotting.LiveScatter' imported but unused
core.py:19: '.mpl_plotting.LivePlot' imported but unused
core.py:19: '.mpl_plotting.LiveGrid' imported but unused
core.py:19: '.mpl_plotting.LiveFitPlot' imported but unused
core.py:19: '.mpl_plotting.LiveRaster' imported but unused
core.py:19: '.mpl_plotting.LiveMesh' imported but unused
```

@danielballan @tacaswell  - Mind if I clean these up and make the matplotlib presence truly optional?

### PR DESCRIPTION
**Pull request by pavoljuhas**
Always import plotting callbacks from `callbacks.mpl_plotting`.
Add test to ensure `bluesky.callbacks` import without matplotlib.

Fixes #1103.

### CODE REVIEW

**Comment by danielballan** [comment]
@pavoljuhas I think #1107 is one way to add (imperfect but IMO good enough) back-compatibility to this approach. If you and @tacaswell think #1107 is reasonable, we should add my commits to this PR here because you cover some niceties (that test script and documentation updates) that I missed.

**Comment by danielballan** [comment]
> I tried similar approach in pavoljuhas@41306b2,

Oh, that is a nice, tidy approach.

> Can we perhaps adopt that version after putting in a more detailed warning from #1107?

Sounds good to me.

> Is there any need to have them there as well as in `callbacks.__init__`?

For the sake of completeness, I would prefer to make sure these plotting functions are importable from both `callbacks.core` and `callbacks.__init__`. It's not much additional work, and I wouldn't be surprised if someone somewhere is importing `LiveTable` from there. Better safe than sorry.

>  it would be better to warn with `DeprecationWarning`.

It's true that's what Python does, but we follow the SciPy/PyData convention of using `UserWarning` because they are the only kind of warning displayed by default, and we need to put this in users' faces so that they update their code at the beamlines. You have a technical point, I agree, but I'd appreciate you indulging on convention on this.

**Comment by pavoljuhas** [comment]
@danielballan - I have put back the plotting callbacks as thin wrappers and used your detailed warning.

File: bluesky/callbacks/core.py [hunk 1]
```diff
@@ -3,22 +3,43 @@
...
+def _deprecate_import_name(name):
+    wmsg = (
+        "In a future version of bluesky, {} will not be importable from "
+        "bluesky.callbacks.core or bluesky.callbacks. Instead, import it from "
+        "bluesky.callbacks.mpl_plotting. This change allows other callbacks, "
+        "unrelated to matplotlib, to be imported and used without importing "
+        "matplotlib.pyplot or configuring a DISPLAY."
+    ).format(name)
+    def f(*args, **kwargs):
+        from warnings import warn
```
[the remark this answers is not in the document]
Comment (pavoljuhas) [reply bluesky/callbacks/core.py]: right, good catch.

**Comment by danielballan** [comment]
Power-cycled...

**Comment by danielballan** [comment]
Thanks again for this detail-oriented work, @pavoljuhas. This will definitely make my life easier when I'm assisting remotely and don't want

**Comment by pavoljuhas** [comment]
@danielballan - sure you are welcome.  In the meantime I found that `export MPLBACKEND=agg` can help with missing DISPLAY too.

**Comment by danielballan** [comment]
Yes, thanks. That's what I've doing but it usually goes like:

```
bsui
# Darn.
bsui --matplotlib=Agg
```

:- D

**Pull request merged**